src/authorization/accessCheckers/GroupAccessChecker.ts

```diff
@@ -0,0 +1,57 @@
+import type { Term } from 'n3';
+import type { Credentials } from '../../authentication/Credentials';
+import type { Representation } from '../../ldp/representation/Representation';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import type { RepresentationConverter } from '../../storage/conversion/RepresentationConverter';
+import { fetchDataset } from '../../util/FetchUtil';
+import { promiseAny } from '../../util/PromiseUtil';
+import { readableToQuads } from '../../util/StreamUtil';
+import { ACL, VCARD } from '../../util/Vocabularies';
+import type { AccessCheckerArgs } from './AccessChecker';
+import { AccessChecker } from './AccessChecker';
+
+export class GroupAccessChecker extends AccessChecker {
+  private readonly converter: RepresentationConverter;
+
+  public constructor(converter: RepresentationConverter) {
+    super();
+    this.converter = converter;
+  }
+
+  public async canHandle(): Promise<void> {
+    // It can always handle the input
+  }
+
+  public async handle({ acl, rule, agent }: AccessCheckerArgs): Promise<boolean> {
+    // Keep only unique values
+    const groups = [ ...new Set(acl.getObjects(rule, ACL.agentGroup, null)) ];
+
+    return await promiseAny(groups.map(async(group: Term): Promise<boolean> => this.isMemberOfGroup(agent, group)));
+  }
+
+  /**
+   * Checks if the given agent is member of a given vCard group.
+   * @param agent - Credentials of agent that needs access.
+   * @param group - URL of the vCard group that needs to be checked.
+   *
+   * @returns If the agent is member of the given vCard group.
+   */
+  private async isMemberOfGroup(agent: Credentials, group: Term): Promise<boolean> {
+    const groupDocument: ResourceIdentifier = { path: /^[^#]*/u.exec(group.value)![0] };
+
+    // Fetch the required vCard group file
+    let dataset: Representation;
+    try {
+      dataset = await fetchDataset(groupDocument.path, this.converter);
+    } catch {
+      /**
+       * If an error is thrown, we don't want this function to propagate it,
+       * because of the semantics of promiseAny.
+       */
```

Comment:
```suggestion
      /**
       * If an error is thrown, we don't want this function to propagate it,
       * because of the semantics of promiseAny.
       */
```

(This component does not know whether it will be used with `promiseAny`.)